tests/sessions_tests/tests.py

```diff
@@ -311,6 +311,17 @@ def test_decode(self):
         encoded = self.session.encode(data)
         self.assertEqual(self.session.decode(encoded), data)
 
+    def test_decode_legacy(self):
+        """
+        Before Django 3.1, encoding was not using signing infrastructure.
+        We should still be able to decode older format until Django 4.0.
+        """
+        legacy_encoded = (
```

Comment:
I think this test should set `SECRET_KEY` to a specific fixed value since `legacy_encoded` depends on that exact value of `SECRET_KEY`. (Granted the chances that we are changing the secret key in the test settings is rather slim, but I'd still feel better if this code does not break if one does).